### PR TITLE
Frontend uc500 510 task rapport

### PR DESCRIPTION
**Pull request by gubelli**
Open issue #24

### CODE REVIEW

File: Frontend/src/app/modification/modification.component.html [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+      <div class="mat-step-navigation">
+        <button mat-button matStepperPrevious>Back</button>
+        <button mat-button matStepperNext>Next</button>
+      </div>
+    </form>
+  </mat-step>
+  <mat-step *ngIf="stepConfiguration.step2.state || stepConfiguration.step3.state || stepConfiguration.step4.state">
+    <ng-template matStepLabel>Confirmation</ng-template>
+    You are now done.
+    <div class="mat-step-navigation">
```
Comment (Alucard1766) [inline Frontend/src/app/modification/modification.component.html]: Change Text to "check ..."

File: Frontend/src/app/modification/modification.component.ts [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+    this.bomFormGroup = this.formBuilder.group({
+      bomPartCtrl: ['', Validators.required]
+    });
+    this.partCatalogFormGroup = this.formBuilder.group({
+      partCatalogCtrl: ['', Validators.required]
+    });
+    this.machineFormGroup = this.formBuilder.group({
+      newValueCtrl: ['', Validators.required]
+    });
+    this.methodFormGroup.reset();
```
Comment (Alucard1766) [inline Frontend/src/app/modification/modification.component.ts]: Make stepper controls untouched. Now after sending modification, the Input is red

File: Frontend/src/app/modification/modification.component.html [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+      <mat-error *ngIf="methodFormGroup.get('methodSelectCtrl').errors &&
+            (methodFormGroup.get('methodSelectCtrl').dirty || methodFormGroup.get('methodSelectCtrl').touched)">
+        <div>Operation method is required</div>
+      </mat-error>
+      <div>
+        <button mat-button matStepperNext>Next</button>
+      </div>
+    </form>
+  </mat-step>
+  <mat-step [stepControl]="bomFormGroup" *ngIf="stepConfiguration.step2.state">
```
[the remark this answers is not in the document]
Comment (gubelli) [reply Frontend/src/app/modification/modification.component.html]: Also renamed Choose shored part - Chosse shored part from bom

File: Frontend/src/app/modification/modification.component.ts (hunk 2 above)
Comment (gubelli) [reply Frontend/src/app/modification/modification.component.ts]: It is not possible for select of method - Switching index of stepper marks it as touched

File: Frontend/src/app/services/taskrapport.service.ts [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+import 'rxjs/add/observable/throw';
+import {BASEURL} from './api.connection';
+
+
+
+@Injectable()
+export class TaskrapportService {
+  baseUrl = BASEURL + '/TaskRapports/' ;
+  jsonHeaders = new HttpHeaders().set('Content-Type', 'application/json');
+  xmlHeaders = new HttpHeaders().set('Content-Type', 'application/xml');
```
[the remark this answers is not in the document]
Comment (gubelli) [reply Frontend/src/app/services/taskrapport.service.ts]: Also extracted XML header as const

File: Frontend/src/app/taskrapport/taskrapport-detail.component.ts [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+  templateUrl: './taskrapport-detail.component.html',
+  styleUrls: ['./taskrapport-detail.component.css']
+})
+
+export class TaskrapportDetailComponent implements OnInit {
+  serviceCases: ServiceCase[];
+  selectedServiceCase: ServiceCase;
+  creation = false;
+  editable = false;
+  modification: boolean;
```
[the remark this answers is not in the document]
Comment (gubelli) [reply Frontend/src/app/taskrapport/taskrapport-detail.component.ts]: I will do it on a separate branche for the complete frontend

**Pull request merged**